tools/rosgraph/src/rosgraph/xmlrpc.py

```diff
@@ -93,6 +93,9 @@ def __init__(self, addr, log_requests=1):
         # to True to allow quick restart on the same port.  This is equivalent 
         # to calling setsockopt(SOL_SOCKET,SO_REUSEADDR,1)
         self.allow_reuse_address = True
+        # Increase request_queue_size to handle issues with many simultaneous
+        # connections in OSX 10.11
+        self.request_queue_size = 128
```

Comment:
In short, I picked 128 because it is the max on OSX and Ubuntu 14.04
(socket.SOMAXCONN gives OS limit).

I suppose it is hard to say if a smaller queue size would eliminate the
problem since I can't say exactly what is causing the problem on my system
vs older OSX builds or Ubuntu.



On Thu, Jul 28, 2016 at 6:15 PM, Dirk Thomas <notifications@github.com>
wrote:

> In tools/rosgraph/src/rosgraph/xmlrpc.py
> <https://github.com/ros/ros_comm/pull/850#discussion_r72713214>:
>
> > @@ -93,6 +93,9 @@ def __init__(self, addr, log_requests=1):
> >          # to True to allow quick restart on the same port.  This is equivalent
> >          # to calling setsockopt(SOL_SOCKET,SO_REUSEADDR,1)
> >          self.allow_reuse_address = True
> > +        # Increase request_queue_size to handle issues with many simultaneous
> > +        # connections in OSX 10.11
> > +        self.request_queue_size = 128
>
> Can you please elaborate why you chose 128.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ros/ros_comm/pull/850/files/7a9bb8f4f188fe7826e1775a88f90fd9260894b3#r72713214>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAIDam_wNt-LSbmaG5fjddDgz03U_Y61ks5qaSodgaJpZM4JWRWH>
> .
>